File: data/readable_format/fortran/file1441.f90
! Define the subroutine with a fix for the 'inv' declaration
subroutine ts_calc(xflow, Tt, Pt, kappa, r, a, Ts, icase)
    implicit none
    integer :: icase, i, inv ! 'inv' variable declaration added here
    double precision :: xflow, Tt, Pt, Ts, kappa, r, f, df, a, expon, Ts_old, C, TtzTs
    double precision :: deltaTs, TtzTs_crit, Qred_crit, Qred, h1, h2, h3
    expon = -0.5d0 * (kappa + 1.d0) / (kappa - 1.d0)
    C = dsqrt(2.d0 / r * kappa / (kappa - 1.d0))
    Ts_old = Tt
    if (xflow .lt. 0d0) then
        inv = -1
    else
        inv = 1
    endif
    if (dabs(xflow) .le. 1e-9) then
        Ts = Tt
        return
    endif
    Qred = abs(xflow) * dsqrt(Tt) / (a * Pt)
    Ts = Tt / (1 + (Qred**2 / C**2))
    if (icase .eq. 0) then
        TtzTs_crit = (kappa + 1.d0) / 2.d0
    else
        TtzTs_crit = (1d0 + (kappa - 1.d0) / (2.d0 * kappa))
    endif
    Qred_crit = C * (TtzTs_crit)**expon * (TtzTs_crit - 1.d0)**0.5d0
    if (Qred .ge. Qred_crit) then
        Ts = Tt / TtzTs_crit
        return
    endif
    i = 0
    do
        i = i + 1
        TtzTs = Tt / Ts
        h1 = TtzTs - 1.d0
        h2 = dsqrt(h1)
        h3 = TtzTs**expon
        f = C * h2 * h3
        df = f * (expon + 0.5d0 * TtzTs / h1) / Ts
        f = Qred - f
        deltaTs = -f / df
        Ts = Ts + deltaTs
        if ((dabs(Ts - Ts_old) / Ts_old) .le. 1.E-8 .or. dabs(Ts - Ts_old) .le. 1.E-10) then
            exit
        else if (i .gt. 20) then
            Ts = 0.9 * Tt
            exit
        endif
        Ts_old = Ts
    end do
    return
end subroutine ts_calc

! Main program to test the subroutine
program test_ts_calc
    implicit none
    double precision :: xflow, Tt, Pt, kappa, r, a, Ts
    integer :: icase

    ! Test parameters
    xflow = 0.5
    Tt = 300.0
    Pt = 101325.0
    kappa = 1.4
    r = 287.0
    a = 340.29
    icase = 0

    ! Call the subroutine
    call ts_calc(xflow, Tt, Pt, kappa, r, a, Ts, icase)

    ! Output the result
    print *, "Ts = ", Ts
end program test_ts_calc

File: data/readable_format/cpp/file1441.cpp
#include <iostream>
#include <cmath>
#include <cassert>

// Define the ts_calc function
void ts_calc(double xflow, double Tt, double Pt, double kappa, double r, double a, double& Ts, int icase) {
    const double expon = -0.5 * (kappa + 1.0) / (kappa - 1.0);
    const double C = sqrt(2.0 / r * kappa / (kappa - 1.0));
    double Ts_old = Tt;
    int inv = (xflow < 0.0) ? -1 : 1;

    if(std::abs(xflow) <= 1e-9) {
        Ts = Tt;
        return;
    }

    double Qred = std::abs(xflow) * sqrt(Tt) / (a * Pt);
    Ts = Tt / (1 + (Qred * Qred / (C * C)));

    double TtzTs_crit;
    if(icase == 0) {
        TtzTs_crit = (kappa + 1.0) / 2.0;
    } else {
        TtzTs_crit = 1.0 + (kappa - 1.0) / (2.0 * kappa);
    }

    double Qred_crit = C * std::pow(TtzTs_crit, expon) * sqrt(TtzTs_crit - 1.0);
    if(Qred >= Qred_crit) {
        Ts = Tt / TtzTs_crit;
        return;
    }

    int i = 0;
    do {
        i++;
        double TtzTs = Tt / Ts;
        double h1 = TtzTs - 1.0;
        double h2 = sqrt(h1);
        double h3 = pow(TtzTs, expon);
        double f = C * h2 * h3;
        double df = f * (expon + 0.5 * TtzTs / h1) / Ts;
        f = Qred - f;
        double deltaTs = -f / df;
        Ts = Ts + deltaTs;

        if ((std::abs(Ts - Ts_old) / Ts_old <= 1E-8) || (std::abs(Ts - Ts_old) <= 1E-10)) {
            break;
        } else if (i > 20) {
            Ts = 0.9 * Tt;
            break;
        }

        Ts_old = Ts;
    } while (true);
}

// Main function to test the ts_calc function
int main() {
    double xflow = 0.5;
    double Tt = 300.0;
    double Pt = 101325.0;
    double kappa = 1.4;
    double r = 287.0;
    double a = 340.29;
    int icase = 0;
    double Ts = 0.0;

    // Call the function
    ts_calc(xflow, Tt, Pt, kappa, r, a, Ts, icase);

    // Output the result
    std::cout << "Ts = " << Ts << std::endl;

    // Basic assertion to ensure Ts was calculated
    assert(Ts != 0.0);

    return 0;
}